### ISSUE TITLE
Remove calls to MD layer functions from X.509

### ISSUE THREAD

**Issue by mpg** [ARMmbed/mbedtls#5158]
Currently calls to PSA for hashing in X.509 are conditional to `MBEDTLS_USE_PSA_CRYPTO`. Since we're now auto-enabling that, we don't need the alternative code path any more.

This task is to replace two instances of:
```
#if defined(MBEDTLS_USE_PSA_CRYPTO)
/* code using psa_hash */
#else
/* code using mbedtls_md */
#endif
```
with:
```
/* code using psa_hash */
```

While at it, the PSA version, which currently uses the multi-part API (with `psa_hash_operation_t`) can be simplified to use the single-part API `psa_hash_compute()`.

- [ ] replaced in `x509_crt_check_signature()`
- [ ] replaced in `mbedtls_x509write_csr_der_internal()`

Depends on: #5156 (to be able to use PSA without #ifdefs).
See also: #5157 

Note: this task is only about MD functions. PK functions may require more care and will be handled as a separate task.

**Comment by mpg** [comment]
Closing as this no longer makes sense with the change of strategy (`USE_PSA_CRYPTO` will remain optional for now).
